scheduler/scheduler.go

```diff
@@ -281,13 +291,32 @@ func generateContainerHostIntents(d distro.Distro, newContainersNeeded int) ([]h
 	return containerHostIntents, nil
 }
 
+func getDockerOptionsFromProviderSettings(settings *map[string]interface{}) (*host.DockerOptions, error) {
```

Comment:
I'm not sure I'm onboard with the move from cloud.ContainerSettings to host.DockerOptions. I remember being pretty confident in the old design. 

Additionally, would it be possible to keep all of these arbitrary settings isolated in provider settings rather than having to manage the mapstructure use in multiple cases?

Reply:
We can talk about it. Basically we can't use cloud.ContainerSettings for the host document so now there's an identical struct floating around. I can extend cloud.ContainerSettings to cover all the parameters I need and just pass every parameter from the host over, but in my opinion it's messy to have two identical structs and we should just consolidate.

Reply:
I assume by your second comment you mean to move this function into provider rather than having it in the scheduler?

Reply:
Another reason to use the new struct is because we need to persist it to the db. It's true that we could shove those in the provider settings in the distro, but that feels wrong. It does feel weird to be in the host data model and not cloud package, but I think it's cleaner than the alternatives

Reply:
Aren't we saving the struct into the database as a *map[string]interface{] and then using mapstruct, or did I confuse that?

Reply:
The information from the struct is saved in the host in the Distro.ProviderSettings as a *map[string]interface{}, which is what John is referring to. If we were to put the container information into providerSettings in scheduler and host_create, the key values would be constants across packages which seems messier than a struct.

Reply:
Also at the moment we only use the mapstruct to decode one parameter, and we manually check for the other pieces in provisioning_create_host:isImageBuilt.